updating/understanding-openshift-upgrade-duration.adoc

```diff
@@ -0,0 +1,96 @@
+====
+The CVO process does not restart the workloads.
+====
+
+=== Machine Config Operator
+The Machine Config Operator (MCO) applies a new machine configuration to each control plane and worker node. During this process, the MCO performs the following actions on each node of the cluster:
+
+. Cordon and drain all nodes
+. Update the operating system (OS)
+. Node reboot
+. Uncordon all nodes and schedule workloads on the node
+
+[NOTE]
+====
+When a node is cordoned, it is unavailable to schedule a workload.
+====
+
+The control plane node upgrade duration is predictable and oftentimes shorter than worker nodes, because the control plane workloads are tuned for graceful updates and quick drains. You can upgrade the worker nodes in parallel by setting the `maxUnavailable` field to greater than 1 in the machine configuration pool. The MCO cordons the number of nodes specified in `maxUnavailable` and marks them unavailable for upgrade.
+
+The time to complete this process depends on many factors including the node and infrastructure configuration. This process might take 5 minutes to complete per node.
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../architecture/architecture-installation.html#update-service-overview_architecture-installation[About the OpenShift Update Service]
+
+==== Pod disruption budget
+You must consider pod disruption budgets and pod anti-affinity when you configure `maxUnavailable` field in the machine configuration pool.
+
+For example, if there is a cluster with three worker nodes, you must configure the disruption budget such that it has minimum two replicas be always running. When you configure pod anti-affinity, you must specify that two replicas should never run on the same node. To avoid pod scheduling issues, you must ensure the availability of minimum two worker nodes.
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../nodes/pods/nodes-pods-configuring.adoc#nodes-pods-configuring-pod-distruption-about[pod disruption budget]
+* xref:../nodes/scheduling/nodes-scheduler-pod-affinity.adoc#nodes-scheduler-pod-affinity-about[pod anti-affinity]
+
+==== Cluster node health
+Before you begin the upgrade, you must ensure that all the nodes are available. Any unavailable nodes can significantly increase the upgrade duration because the node unavailability affects the `maxUnavailable` and pod disruption budgets.
+
+To check the status of nodes, run the following command:
+[source,terminal]
+----
+$ oc get nodes -o wide
+----
+
+==== {op-system-base-full} nodes
+{op-system-base-full} worker nodes require an additional usage of OpenShift-Ansible to update node binary components. This rolling update is similar to the MCO initiated rolling update and therefore it has a similar duration. The time it takes to reboot a particular node varies significantly. In cloud instances, the reboot might take less than a minute whereas in physical bare metal hosts the reboot takes more than 15 minutes.
```

Comment:
> a similar duration
Means 5 min per node? I notice there is only an estimation about the host reboot time, but seems not a whole process duration of rhel node update.

Reply:
According to my limit test experience on rhel upgrade, the host reboot time seems more than 1min even on cloud platform. 
May @gpei could help on this part.

Reply:
@jiajliu hi, I collected several reboot duration log of RHEL machines from our CI testing for your reference, seems it's always about 1~2 minutes on those public IaaS. 

```
Azure:
TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Saturday 11 June 2022  17:12:56 +0800 (0:00:00.804)       0:04:18.453 *********
changed: [10.0.1.9] => {"changed": true, "elapsed": 45, "rebooted": true}
changed: [10.0.1.8] => {"changed": true, "elapsed": 60, "rebooted": true}

TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Sunday 12 June 2022  01:14:47 +0800 (0:00:00.824)       0:04:16.762 ***********
changed: [10.0.1.8] => {"changed": true, "elapsed": 64, "rebooted": true}
changed: [10.0.1.7] => {"changed": true, "elapsed": 74, "rebooted": true}

TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Saturday 04 June 2022  17:09:54 +0800 (0:00:00.879)       0:04:55.928 ********* 
changed: [10.0.1.7] => {"changed": true, "elapsed": 79, "rebooted": true}
changed: [10.0.1.8] => {"changed": true, "elapsed": 103, "rebooted": true}


AWS:
TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Saturday 11 June 2022  16:50:32 +0800 (0:00:00.677)       0:04:06.007 *********
changed: [ip-10-0-60-226.us-east-2.compute.internal] => {"changed": true, "elapsed": 75, "rebooted": true}
changed: [ip-10-0-53-20.us-east-2.compute.internal] => {"changed": true, "elapsed": 75, "rebooted": true}

TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Wednesday 15 June 2022  00:59:58 +0800 (0:00:00.611)       0:03:31.118 ********
changed: [ip-10-0-51-74.us-east-2.compute.internal] => {"changed": true, "elapsed": 61, "rebooted": true}

TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Saturday 11 June 2022  09:00:15 +0800 (0:00:00.620)       0:04:09.054 *********
changed: [ip-10-0-48-116.us-east-2.compute.internal] => {"changed": true, "elapsed": 75, "rebooted": true}
changed: [ip-10-0-59-128.us-east-2.compute.internal] => {"changed": true, "elapsed": 90, "rebooted": true}


GCP:
TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Sunday 12 June 2022  16:21:35 +0800 (0:00:00.797)       0:08:53.077 ***********
changed: [10.0.128.6] => {"changed": true, "elapsed": 62, "rebooted": true}
changed: [10.0.128.5] => {"changed": true, "elapsed": 65, "rebooted": true}

TASK [openshift_node : Reboot the host and wait for it to come back] ***********
Sunday 05 June 2022  08:55:21 +0800 (0:00:00.825)       0:09:03.885 *********** 
changed: [10.0.128.6] => {"changed": true, "elapsed": 62, "rebooted": true}
changed: [10.0.128.5] => {"changed": true, "elapsed": 62, "rebooted": true}
```

Reply:
> bare metal hosts the reboot takes more than 15 minutes.
Thanks @gpei . And for bare metal, should you or Pedro Amoedo could help check this?

Reply:
We didn't test RHEL scaling up on Packet, for baremetal platform, we're using the RHEL VM instances on OSP for baremetal platform coverage. 
But I guess @pamoedom could help providing some insights on how long will a RHEL physical bare metal host reboot take.

Reply:
> We didn't test RHEL scaling up on Packet, for baremetal platform, we're using the RHEL VM instances on OSP for baremetal platform coverage. But I guess @pamoedom could help providing some insights on how long will a RHEL physical bare metal host reboot usually take.

Well, first of all, I can only provide estimation times with RHCOS and not RHEL (RHCOS is lighter), secondly, on real BM, you should take into account the extra boot time that each model has (BIOS, post, iPXE boot timeout, etc), with all of that into account and taking medium instance size as reference, I can estimate about 5min when the boot is clean without any disk issue.

Regards.

Reply:
This thread discussion is moved to https://github.com/openshift/openshift-docs/pull/46241#discussion_r917665312.